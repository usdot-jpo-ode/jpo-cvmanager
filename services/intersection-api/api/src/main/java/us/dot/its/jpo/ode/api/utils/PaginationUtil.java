package us.dot.its.jpo.ode.api.utils;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;

public class PaginationUtil {

    public static <T> ResponseEntity<List<T>> createResponseEntityWithPaginationHeaders(Page<T> page) {
        HttpHeaders headers = new HttpHeaders();
        headers.add("X-Total-Count", String.valueOf(page.getTotalElements()));
        headers.add("X-Total-Pages", String.valueOf(page.getTotalPages()));
        headers.add("X-Has-More", String.valueOf(page.hasNext()));
        return new ResponseEntity<List<T>>(page.getContent(), headers, HttpStatus.OK);
    }
}