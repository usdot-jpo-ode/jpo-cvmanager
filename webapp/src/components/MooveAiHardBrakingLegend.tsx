import React from 'react'
import { Typography } from '@mui/material'

const MooveAiHardBrakingLegend = () => {
  return (
    <div className="hardBrakingLegend">
      <Typography fontFamily="Arial, Helvetica, sans-serif" fontSize="small" align="center">
        <b>Hard Braking Color Guide</b>
      </Typography>
      <div style={gridStyle}>
        <div style={gridRowStyle}>
          <div style={{ width: '20px', height: '20px', backgroundColor: 'rgb(0, 255, 0)', marginRight: '10px' }}></div>
          <Typography fontFamily="Arial, Helvetica, sans-serif" fontSize="small">
            0-249 Hard Brakes
          </Typography>
        </div>
        <div style={gridRowStyle}>
          <div
            style={{ width: '20px', height: '20px', backgroundColor: 'rgb(255, 255, 0)', marginRight: '10px' }}
          ></div>
          <Typography fontFamily="Arial, Helvetica, sans-serif" fontSize="small">
            250-499 Hard Brakes
          </Typography>
        </div>
        <div style={gridRowStyle}>
          <div
            style={{ width: '20px', height: '20px', backgroundColor: 'rgb(255, 165, 0)', marginRight: '10px' }}
          ></div>
          <Typography fontFamily="Arial, Helvetica, sans-serif" fontSize="small">
            500-749 Hard Brakes
          </Typography>
        </div>
        <div style={gridRowStyle}>
          <div style={{ width: '20px', height: '20px', backgroundColor: 'rgb(255, 0, 0)', marginRight: '10px' }}></div>
          <Typography fontFamily="Arial, Helvetica, sans-serif" fontSize="small">
            750+ Hard Brakes
          </Typography>
        </div>
        <div style={gridBottomStyle}></div>
      </div>
      <Typography fontFamily="Arial, Helvetica, sans-serif" fontSize="small" style={{ marginLeft: '10px' }}>
        Determined by a weekly average
      </Typography>
    </div>
  )
}

const gridStyle: React.CSSProperties = {
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'flex-start',
  marginTop: '10px',
  marginLeft: '10px',
}

const gridRowStyle: React.CSSProperties = {
  display: 'flex',
  alignItems: 'center',
  marginBottom: '5px',
}

const gridBottomStyle: React.CSSProperties = {
  display: 'flex',
  alignItems: 'center',
  marginBottom: '10px',
}

export default MooveAiHardBrakingLegend
