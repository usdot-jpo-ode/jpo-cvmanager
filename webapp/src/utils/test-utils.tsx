/**
 * This function searches for the every somewhat random/chaotic class name and property that cause snapshot tests to be inconsistent.
 * This current list includes MUI classes (css-*) and aria-invalid attributes.
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceChaoticIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container: HTMLElement) {
  const props = [
    {
      selector: 'class',
      updateFunc: (val: string) => val.replace(/css-[0-9a-z]*?-/g, 'css-mocked-'),
    },
  ]

  container.querySelectorAll('*').forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)))
      }
    })
  })
  container.querySelectorAll('input[aria-invalid]').forEach((item) => {
    item.removeAttribute('aria-invalid')
  })
  return container
}

import React from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider'
import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs'
import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'

// Extend dayjs with timezone and UTC plugins
dayjs.extend(utc)
dayjs.extend(timezone)

// Mock component to set timezone
const MockLocalizationProvider = ({ children }) => {
  dayjs.tz.setDefault('America/Denver')
  return <LocalizationProvider dateAdapter={AdapterDayjs}>{children}</LocalizationProvider>
}

export { replaceChaoticIds, MockLocalizationProvider }
