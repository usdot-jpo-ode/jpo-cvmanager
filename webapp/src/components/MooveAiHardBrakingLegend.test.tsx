import React from 'react'
import { render } from '@testing-library/react'
import MooveAiHardBrakingLegend from './MooveAiHardBrakingLegend'
import { replaceChaoticIds } from '../utils/test-utils'

it('should take a snapshot', () => {
  const { container } = render(<MooveAiHardBrakingLegend />)

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
