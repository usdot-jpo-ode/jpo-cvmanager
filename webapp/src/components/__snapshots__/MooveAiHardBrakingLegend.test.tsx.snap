// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<div>
  <div
    class="hardBrakingLegend"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-mocked-MuiTypography-root"
      style="--Typography-textAlign: center;"
    >
      <b>
        Hard Braking Color Guide
      </b>
    </p>
    <div
      style="display: flex; flex-direction: column; align-items: flex-start; margin-top: 10px; margin-left: 10px;"
    >
      <div
        style="display: flex; align-items: center; margin-bottom: 5px;"
      >
        <div
          style="width: 20px; height: 20px; background-color: rgb(0, 255, 0); margin-right: 10px;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          0-249 Hard Brakes
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin-bottom: 5px;"
      >
        <div
          style="width: 20px; height: 20px; background-color: rgb(255, 255, 0); margin-right: 10px;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          250-499 Hard Brakes
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin-bottom: 5px;"
      >
        <div
          style="width: 20px; height: 20px; background-color: rgb(255, 165, 0); margin-right: 10px;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          500-749 Hard Brakes
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin-bottom: 5px;"
      >
        <div
          style="width: 20px; height: 20px; background-color: rgb(255, 0, 0); margin-right: 10px;"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
        >
          750+ Hard Brakes
        </p>
      </div>
      <div
        style="display: flex; align-items: center; margin-bottom: 10px;"
      />
    </div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-mocked-MuiTypography-root"
      style="margin-left: 10px;"
    >
      Determined by a weekly average
    </p>
  </div>
</div>
`;
